src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs

```diff
@@ -148,5 +150,53 @@ private static string GetPropertyDisplayValue(PropertyValue property)
                 return String.Empty;
             }
         }
+
+        public static void CopyMatchingNodesIntoSkeletonDTO(ExecutionGraphComparisonResult destRoot, ExecutionGraphComparisonResult srcRoot)
+        {
+            var srcGraphLookupTable = CreateSkeletonLookupTableFor(srcRoot);
+
+            var queue = new Queue<ExecutionGraphComparisonResult>();
+            queue.Enqueue(destRoot);
+
+            while (queue.Count != 0)
+            {
+                var curNode = queue.Dequeue();
+
+                for (int index = 0; index < curNode.MatchingNodes.Count; ++index)
+                {
+                    var matchingId = curNode.MatchingNodes[index].BaseNode.ID;
+                    var matchingNode = srcGraphLookupTable[matchingId];
+
+                    curNode.MatchingNodes[index] = matchingNode;
+                }
+
+                foreach (var child in curNode.Children)
+                {
+                    queue.Enqueue(child);
+                }
+            }
+        }
+
+        private static Dictionary<int, ExecutionGraphComparisonResult> CreateSkeletonLookupTableFor(ExecutionGraphComparisonResult node)
```

Comment:
Given the name - is this something you could even have be an extension method on ExecutionGraphComparisonResult? So it could just be `result.CreateSkeletonLookupTable()`